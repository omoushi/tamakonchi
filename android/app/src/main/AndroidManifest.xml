<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.tamakonchi">

    <uses-permission android:name="android.permission.INTERNET" />

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
      <meta-data
        android:name="io.fabric.ApiKey"
        android:value="c9e466da83a5a3f3695fcc0b1f52feb2291cf522" />
    </application>
</manifest>
