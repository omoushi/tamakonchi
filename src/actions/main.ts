import { Action, ActionCreator } from "redux";

export enum ActionType {
  BREAK_UP = 'BREAK_UP',
  NEGLECT = 'NEGLECT',
  LOSE_JOB = 'LOSE_JOB',
  MAKE_WORRY = 'MAKE_WORRY'
}

export const breakUp: ActionCreator<Action> = () => ({ type: ActionType.BREAK_UP });

export const neglect: ActionCreator<Action> = () => ({ type: ActionType.NEGLECT });

export const loseJob: ActionCreator<Action> = () => ({ type: ActionType.LOSE_JOB });

export const makeWorry: ActionCreator<Action> = () => ({ type: ActionType.MAKE_WORRY });
