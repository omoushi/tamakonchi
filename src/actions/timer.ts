import {Action, ActionCreator} from "redux";

export enum ActionType {
  MORE_TOOL = 'MORE_TOOL',
}

export const moreTool: ActionCreator<Action> = () => ({type: ActionType.MORE_TOOL});
