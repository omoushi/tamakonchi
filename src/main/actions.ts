import { Action, ActionCreator } from "redux";

export enum ActionType {
  TAKE_CARE = 'TAKE_CARE',
  NOT_TAKE_CARE = 'NOT_TAKE_CARE',
  BREAK_UP = 'BREAK_UP',
  NEGLECT = 'NEGLECT'
}

export const takeCare: ActionCreator<Action> = () => ({ type: ActionType.TAKE_CARE });

export const notTakeCare: ActionCreator<Action> = () => ({ type: ActionType.NOT_TAKE_CARE });

export const breakUp: ActionCreator<Action> = () => ({ type: ActionType.BREAK_UP });

export const neglect: ActionCreator<Action> = () => ({ type: ActionType.NEGLECT });
