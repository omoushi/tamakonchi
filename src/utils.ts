export function randomIn<E>(array: E[]): E {
  return array[Math.floor(Math.random() * array.length)];
}