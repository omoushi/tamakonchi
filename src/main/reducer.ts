import { MainState, MainStateEvent } from "./interface";
import { Action, Reducer } from "redux";
import { ActionType } from "./actions";

export enum Statuses { STAGE }

export enum Stage {
  EGG = 'たまご', LIVING = 'いきてる', DIED = 'しんでる'
}

export enum Situation {
  NORMAL = '通常', DEAD = 'しんだ', BrokenUp = '別れた'
}

const HATCHING: MainStateEvent = { stage: Stage.LIVING };
const DIEING: MainStateEvent = { stage: Stage.DIED };
const BREAK_UP: MainStateEvent = { situation: Situation.BrokenUp }
const NEGLECT: MainStateEvent = { situation: Situation.DEAD }

const breakUp = (state: MainState): MainStateEvent => (state.situation === Situation.NORMAL ? BREAK_UP : {});
const neglect = (state: MainState): MainStateEvent => (state.situation === Situation.NORMAL ? NEGLECT : {})
const takingCare = (state: MainState): MainStateEvent => (state.stage === Stage.EGG ? HATCHING : {});
const notTakingCare = (state: MainState): MainStateEvent => {
  const changeTarget: Statuses = Statuses.STAGE
  if (changeTarget === Statuses.STAGE && state.stage !== Stage.DIED) {
    return DIEING
  } else {
    return {}
  }
};

const initialState: MainState = {
  stage: Stage.EGG,
  situation: Situation.NORMAL
};

export const main: Reducer = (state: MainState = initialState, action: Action<ActionType>): MainState => {
  const stateEventCreator = {
    [ActionType.TAKE_CARE]: takingCare,
    [ActionType.NOT_TAKE_CARE]: notTakingCare,
    [ActionType.BREAK_UP]: breakUp,
    [ActionType.NEGLECT]: neglect
  }[action.type];
  const stateEvent = stateEventCreator ? stateEventCreator(state) : {};
  return { ...state, ...stateEvent };
};