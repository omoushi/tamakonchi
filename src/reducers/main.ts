import { Action, Reducer } from "redux";

import { MainState, MainStateEvent } from "../types/main";
import { ActionType } from "../actions/main";

export enum Stage {
  NORMAL = '通常',
  DEAD = 'しんだ',
  BROKEN_UP = '別れた'
}

export enum Situation {
  EMPTY = 'なし',
  NO_JOB = '仕事がない',
  LINE = 'LINEが来る'
}

const NO_EVENT: MainStateEvent = { }
const BREAK_UP: MainStateEvent = { stage: Stage.BROKEN_UP }
const NEGLECT: MainStateEvent = { stage: Stage.DEAD }
const LOSE_JOB: MainStateEvent = { situation: Situation.NO_JOB }
const LINE: MainStateEvent = { situation: Situation.LINE }

const breakUp = (state: MainState): MainStateEvent => (state.stage === Stage.NORMAL ? BREAK_UP : NO_EVENT);
const neglect = (state: MainState): MainStateEvent => (state.stage === Stage.NORMAL ? NEGLECT : NO_EVENT)
const loseJob = (state: MainState): MainStateEvent => {
  if (state.stage === Stage.NORMAL && state.situation === Situation.EMPTY) {
    return LOSE_JOB
  } else {
    return NO_EVENT
  }
}
const worry = (state: MainState): MainStateEvent => {
  if (state.stage === Stage.NORMAL && state.situation === Situation.EMPTY) {
    return LINE
  } else {
    return NO_EVENT
  }
}

const initialState: MainState = {
  stage: Stage.NORMAL,
  situation: Situation.EMPTY
};

export const main: Reducer = (state: MainState = initialState, action: Action<ActionType>): MainState => {
  const stateEventCreator = {
    [ActionType.BREAK_UP]: breakUp,
    [ActionType.NEGLECT]: neglect,
    [ActionType.LOSE_JOB]: loseJob,
    [ActionType.MAKE_WORRY]: worry
  }[action.type];
  const stateEvent = stateEventCreator ? stateEventCreator(state) : {};
  return { ...state, ...stateEvent };
};