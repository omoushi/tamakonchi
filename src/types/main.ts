import { Action } from "redux"

import { Situation, Stage } from "../reducers/main";

export type MainState = {
  stage: Stage,
  situation: Situation
}

export type MainStateEvent = {
  stage?: Stage
  situation?: Situation
}

export type MainEvents = {
  breakUp(): Action
  neglect(): Action
  loseJob(): Action
  makeWorry(): Action
}