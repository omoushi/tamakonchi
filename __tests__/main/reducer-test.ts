import {Health, main, Stage, Statuses} from "../../src/main/reducer";
import {notTakeCare, takeCare} from "../../src/main/actions";
import {randomIn} from "../../src/utils";

jest.mock('../../src/utils');
const mockRandomIn = <jest.Mock<Statuses>>randomIn;


const statePatterns = [
  undefined,
  {stage: Stage.EGG, health: Health.GOOD},
  {stage: Stage.EGG, health: Health.BAD},
  {stage: Stage.LIVING, health: Health.GOOD},
  {stage: Stage.LIVING, health: Health.BAD},
  {stage: Stage.DIED, health: Health.UNKNOWN}
];

describe('takeCare', () => {
  statePatterns.forEach(state => {
    it(`state: ${JSON.stringify(state)}`, () => {
      expect(main(state, takeCare())).toMatchSnapshot()
    });
  });
});

describe('notTakeCare', () => {


  describe('状態が変わる場合', () => {
    mockRandomIn.mockReturnValue(Statuses.STAGE);
    statePatterns.forEach(state => {
      it(`state: ${JSON.stringify(state)}`, () => {
        expect(main(state, notTakeCare())).toMatchSnapshot()
      });
    });
  });

  describe('健康が変わる場合', () => {
    mockRandomIn.mockReturnValue(Statuses.HEALTH);
    statePatterns.forEach(state => {
      it(`state: ${JSON.stringify(state)}`, () => {
        expect(main(state, notTakeCare())).toMatchSnapshot()
      });
    });
  });
});