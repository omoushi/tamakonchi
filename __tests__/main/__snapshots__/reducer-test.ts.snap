// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakUp state: {"stage":"いきてる","situation":"通常"} 1`] = `
Object {
  "situation": "別れた",
  "stage": "いきてる",
}
`;

exports[`breakUp state: {"stage":"いきてる","situation":"通常"} 2`] = `
Object {
  "situation": "別れた",
  "stage": "いきてる",
}
`;

exports[`breakUp state: {"stage":"しんでる","situation":"通常"} 1`] = `
Object {
  "situation": "別れた",
  "stage": "しんでる",
}
`;

exports[`breakUp state: {"stage":"たまご","situation":"通常"} 1`] = `
Object {
  "situation": "別れた",
  "stage": "たまご",
}
`;

exports[`breakUp state: {"stage":"たまご","situation":"通常"} 2`] = `
Object {
  "situation": "別れた",
  "stage": "たまご",
}
`;

exports[`breakUp state: undefined 1`] = `
Object {
  "situation": "別れた",
  "stage": "たまご",
}
`;

exports[`neglect state: {"stage":"いきてる","situation":"通常"} 1`] = `
Object {
  "situation": "しんだ",
  "stage": "いきてる",
}
`;

exports[`neglect state: {"stage":"いきてる","situation":"通常"} 2`] = `
Object {
  "situation": "しんだ",
  "stage": "いきてる",
}
`;

exports[`neglect state: {"stage":"しんでる","situation":"通常"} 1`] = `
Object {
  "situation": "しんだ",
  "stage": "しんでる",
}
`;

exports[`neglect state: {"stage":"たまご","situation":"通常"} 1`] = `
Object {
  "situation": "しんだ",
  "stage": "たまご",
}
`;

exports[`neglect state: {"stage":"たまご","situation":"通常"} 2`] = `
Object {
  "situation": "しんだ",
  "stage": "たまご",
}
`;

exports[`neglect state: undefined 1`] = `
Object {
  "situation": "しんだ",
  "stage": "たまご",
}
`;

exports[`notTakeCare 状態が変わる場合 state: {"stage":"いきてる","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`notTakeCare 状態が変わる場合 state: {"stage":"いきてる","situation":"通常"} 2`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`notTakeCare 状態が変わる場合 state: {"stage":"しんでる","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`notTakeCare 状態が変わる場合 state: {"stage":"たまご","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`notTakeCare 状態が変わる場合 state: {"stage":"たまご","situation":"通常"} 2`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`notTakeCare 状態が変わる場合 state: undefined 1`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`takeCare state: {"stage":"いきてる","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "いきてる",
}
`;

exports[`takeCare state: {"stage":"いきてる","situation":"通常"} 2`] = `
Object {
  "situation": "通常",
  "stage": "いきてる",
}
`;

exports[`takeCare state: {"stage":"しんでる","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "しんでる",
}
`;

exports[`takeCare state: {"stage":"たまご","situation":"通常"} 1`] = `
Object {
  "situation": "通常",
  "stage": "いきてる",
}
`;

exports[`takeCare state: {"stage":"たまご","situation":"通常"} 2`] = `
Object {
  "situation": "通常",
  "stage": "いきてる",
}
`;

exports[`takeCare state: undefined 1`] = `
Object {
  "situation": "通常",
  "stage": "いきてる",
}
`;
